packages/api/lib/rulesHelpers.js

```diff
@@ -113,13 +122,157 @@ async function queueMessageForRule(rule, eventObject, eventSource) {
+  return undefined;
+}
+
+/**
+ * Delete event source mappings for all mappings in the kinesisSourceEvents
+ * @param {Knex} knex - DB client
+ * @param {Object} rule - the rule item
+ * @returns {Promise<Array>} array of responses from the event source deletion
+ */
+async function deleteKinesisEventSources(knex, rule) {
+  const kinesisSourceEvents = [
+    {
+      name: process.env.messageConsumer,
+      eventType: 'arn',
+      type: {
+        arn: rule.arn,
+      },
+    },
+    {
+      name: process.env.KinesisInboundEventLogger,
+      eventType: 'log_event_arn',
+      type: {
+        log_event_arn: rule.log_event_arn,
+      },
+    },
+  ];
+  const deleteEventPromises = kinesisSourceEvents.map(
+    (lambda) => deleteKinesisEventSource(knex, rule, lambda.eventType, lambda.type).catch(
+      (error) => {
+        log.error(`Error deleting eventSourceMapping for ${rule.name}: ${error}`);
+        if (error.code !== 'ResourceNotFoundException') throw error;
+      }
+    )
+  );
+  return await Promise.all(deleteEventPromises);
+}
+
+/**
+ * Delete a rule's SNS trigger
+ * @param {Knex} knex - DB client
+ * @param {Object} rule - the rule item
+ * @returns {Promise} the response from SNS unsubscribe
+ */
+async function deleteSnsTrigger(knex, rule) {
+  // If event source mapping is shared by other rules, don't delete it
+  if (await isEventSourceMappingShared(knex, rule, { arn: rule.arn })) {
+    log.info(`Event source mapping ${rule} with type 'arn' is shared by multiple rules, so it will not be deleted.`);
+    return Promise.resolve();
+  }
+  // delete permission statement
+  const permissionParams = {
+    FunctionName: process.env.messageConsumer,
+    StatementId: `${rule.name}Permission`,
+  };
+  try {
+    await awsServices.lambda().removePermission(permissionParams).promise();
+  } catch (error) {
+    if (isResourceNotFoundException(error)) {
+      throw new ResourceNotFoundError(error);
+    }
+    throw error;
+  }
+  // delete sns subscription
+  const subscriptionParams = {
+    SubscriptionArn: rule.arn,
+  };
+  return awsServices.sns().unsubscribe(subscriptionParams).promise();
+}
+
+/**
+ * Delete rule resources by rule type
+ * @param {Knex} knex - DB client
+ * @param {Object} rule - Rule
+ */
+async function deleteRuleResources(knex, rule) {
+  const type = rule.type;
+  log.info(`Initiating deletion of rule ${JSON.stringify(rule)}`);
+  switch (type) {
+  case 'scheduled': {
+    const targetId = 'lambdaTarget';
+    const name = `${process.env.stackName}-custom-${rule.name}`;
+    await CloudwatchEvents.deleteTarget(targetId, name);
+    await CloudwatchEvents.deleteEvent(name);
+    break;
+  }
+  case 'kinesis': {
+    await deleteKinesisEventSources(knex, rule);
+    break;
+  }
+  case 'sns': {
+    if (rule.enabled === true) {
+      await deleteSnsTrigger(knex, rule);
+    }
+    break;
+  }
+  case 'sqs':
```

Comment:
Can `case 'sqs':` be removed or is it here to be explicit documentation that we do nothing when the `rule.type` is 'sqs'?

Reply:
If we are explicitly documenting that we do nothing for rules of type `sqs` then maybe a test should be added to validate that.